examples/role.ts

```diff
@@ -0,0 +1,23 @@
+import { createClient } from '@clickhouse/client' // or '@clickhouse/client-web'
+
+void (async () => {
+  const client = createClient({
+    role: 'role_name_1',
+  })
+
+  // with a role defined in the client configuration, all queries will use the specified role
+  await client.command({
+    query: `SELECT * FROM SECURED_TABLE`,
+  })
+
+  // one or more roles can be specified in a query as well, to override the role(s) set for the client
+  const rows1 = await client.query({
+    query: `SELECT * FROM VERY_SECURED_TABLE`,
+    format: 'JSONEachRow',
+    role: ['highly_privileged_role'],
+  })
+
+  console.log(await rows1.json())
+
+  await client.close()
+})()
```

Comment:
We aim to have runnable examples; I suggest using the roles bootstrap from tests here.

It is also a good idea to:
- demonstrate expected failures with CH error codes (printing it in the console will suffice so that we can see that 497 ACCESS_DENIED, just like in some other examples, like `abort_request.ts`, `cancel_query.ts`, and probably some others)
- print the output rows like in all other examples (the result set should be consumed, as it holds an unreleased keep-alive socket, waiting for streaming to start; again, a misuse in an example - ResultSets should not be "dangling" in the code)

Reply:
Updated the examples so that they run successfully, consume output rows, and print any error codes.

Example output:
<img width="1532" alt="Screenshot 2024-09-28 at 10 09 35 AM" src="https://github.com/user-attachments/assets/902d08bf-9bba-402d-8ad6-d87289197423">